### PR TITLE
Implement big uint conditional select for modexp

### LINKED ISSUE #134: Implement Big UInt Conditional Select for `modexp`
**Context:** `modexp.yul`

**Description:** 

```solidity
/// @notice Return `rhsPtr` if `cond` is truthy, otherwise return `lhsPtr`.
bigUIntCondSelect(lhsPtr, rhsPtr, cond, nLimbs) -> retPtr
```

`lhsPtr`: The pointer to the MSB of the left operand.
`rhsPtr`: The pointer to the MSB of the right operand.
`cond`: The condition.
`nLimbs`: The number of limbs needed to represent the operands.
`retPtr`: `rhsPtr` if `cond` is truthy, otherwise return `lhsPtr`.

**Recommendation:**

Follow [this implementation](https://github.com/lambdaclass/lambdaworks/blob/main/math/src/unsigned_integer/element.rs#L771).

### PR DESCRIPTION
**Pull request by jpcenteno**
closes #134

### CODE REVIEW

**Review by ilitteri — CHANGES_REQUESTED** [review]
Just one minor detail

**Comment by ilitteri** [comment]
After solving the compilation error, I ran the following tests in a `Playground.yul` file:

```solidity
// BIG UINT COND SELECT TESTS
let borrow1 := 0x00
let borrow2 := 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff

// 1 Limb
let lhsPtr := 0x00
let rhsPtr := 0x20
let orPtr := 0x40
let nLimbs := 1

mstore(lhsPtr, 0x00)
mstore(rhsPtr, 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow1)
console_log(mload(orPtr)) // 0x00

mstore(lhsPtr, 0x00)
mstore(rhsPtr, 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow2)
console_log(mload(orPtr)) // 0x00

// Selects lhs
mstore(lhsPtr, 0x01)
mstore(rhsPtr, 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow1)
console_log(mload(orPtr)) // 0x01 = lhs

// Selects rhs
mstore(lhsPtr, 0x01)
mstore(rhsPtr, 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow2)
console_log(mload(orPtr)) // 0x00 = rhs

// 2 Limbs
let lhsPtr := 0x00
let rhsPtr := 0x40
let orPtr := 0x80
let nLimbs := 2

mstore(lhsPtr, 0x00)
mstore(add(lhsPtr, 0x20), 0x00)
mstore(rhsPtr, 0x00)
mstore(add(rhsPtr, 0x20), 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow1)
console_log(mload(orPtr)) // 0x00
console_log(mload(add(orPtr, 0x20))) // 0x00

mstore(lhsPtr, 0x00)
mstore(add(lhsPtr, 0x20), 0x00)
mstore(rhsPtr, 0x00)
mstore(add(rhsPtr, 0x20), 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow2)
console_log(mload(orPtr)) // 0x00
console_log(mload(add(orPtr, 0x20))) // 0x00

// Selects lhs
mstore(lhsPtr, 0x01)
mstore(add(lhsPtr, 0x20), 0x00)
mstore(rhsPtr, 0x00)
mstore(add(rhsPtr, 0x20), 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow1)
console_log(mload(orPtr)) // 0x01
console_log(mload(add(orPtr, 0x20))) // 0x00

// Selects rhs
mstore(lhsPtr, 0x01)
mstore(add(lhsPtr, 0x20), 0x00)
mstore(rhsPtr, 0x00)
mstore(add(rhsPtr, 0x20), 0x00)
bigUIntCondSelect(lhsPtr, rhsPtr, orPtr, nLimbs, borrow2)
console_log(mload(orPtr)) // 0x00
console_log(mload(add(orPtr, 0x20))) // 0x00
```

with this call
```shell
zksync-era-cli --host localhost --port 8011 call --contract 0x000000000000000000000000000000000000FFFF --function "" --data 00  --private-key 0x850683b40d4a740aa6e745f889a6fdc8327be76e122f5aba645a5b02d0248db8
```

and got this output:
```shell
Call SUCCESS
=== Console Logs:
0
0
1
0
0
0
0
0
1
0
0
0
```

File: precompiles/Modexp.yul [hunk 1]
```diff
@@ -81,6 +81,27 @@ object "ModExp" {
...
+                {
+                    let ptr_lhs_i := add(lhsPtr, offset_i)
+                    let ptr_rhs_i := add(rhsPtr, offset_i)
+                    let ptr_res_i := add(resPtr, offset_i)
+                    let value_lhs_i := mload(ptr_lhs_i)
+                    let value_rhs_i := mload(ptr_rhs_i)
+                    let value_res_i := xor(value_lhs_i, and(mask, xor(value_lhs_i, value_rhs_i))) // a ^ (ct & (a ^ b))
+                    mstore(ptr_res_i, value_res_i)
+                }
+
```
[the remark this answers is not in the document]
Comment (jpcenteno) [reply precompiles/Modexp.yul:104]: Yeah, I messed up after resolving conflicts

**Comment by jpcenteno** [comment]
- [ ] Resolve merge conflicts
- [ ] Run the manual tests again

**Review by ilitteri — APPROVED** [review]

**Pull request merged**